src/QMCWaveFunctions/BsplineFactory/SplineC2C.cpp

```diff
@@ -52,6 +52,109 @@ bool SplineC2C<ST>::write_splines(hdf_archive& h5f)
+  actual number of splined orbitals. This means that in practice rot_mat
+  may be smaller than the number of 'columns' in the coefs array!
+
+      SplineR2R spl_coef layout:
+             ^         | sp1 | ... | spN | pad |
+             |         |=====|=====|=====|=====|
+             |         | c11 | ... | c1N | 0   |
+      basis_set_size   | c21 | ... | c2N | 0   |
+             |         | ... | ... | ... | 0   |
+             |         | cN1 | ... | cNN | 0   |
+             v         |=====|=====|=====|=====|
+                       <------ Nsplines ------>
+
+      SplineC2C spl_coef layout:
+             ^         | sp1_r | sp1_i |  ...  | spN_r | spN_i |  pad  |
+             |         |=======|=======|=======|=======|=======|=======|
+             |         | c11_r | c11_i |  ...  | c1N_r | c1N_i |   0   |
+      basis_set_size   | c21_r | c21_i |  ...  | c2N_r | c2N_i |   0   |
+             |         |  ...  |  ...  |  ...  |  ...  |  ...  |  ...  |
+             |         | cN1_r | cN1_i |  ...  | cNN_r | cNN_i |   0   |
+             v         |=======|=======|=======|=======|=======|=======|
+                       <------------------ Nsplines ------------------>
+
+  NB: For splines (typically) BasisSetSize >> OrbitalSetSize, so the spl_coefs
+  "matrix" is very tall and skinny.
+*/
+template<typename ST>
+void SplineC2C<ST>::applyRotation(const ValueMatrix& rot_mat, bool use_stored_copy)
+{
+  // SplineInst is a MultiBspline. See src/spline2/MultiBspline.hpp
+  const auto spline_ptr = SplineInst->getSplinePtr();
+  assert(spline_ptr != nullptr);
+  const auto spl_coefs      = spline_ptr->coefs;
+  const auto Nsplines       = spline_ptr->num_splines; // May include padding
+  const auto coefs_tot_size = spline_ptr->coefs_size;
+  const auto basis_set_size = coefs_tot_size / Nsplines;
+  assert(OrbitalSetSize == rot_mat.rows());
+  assert(OrbitalSetSize == rot_mat.cols());
+
+  if (!use_stored_copy)
+  {
+    assert(coef_copy_ != nullptr);
+    std::copy_n(spl_coefs, coefs_tot_size, coef_copy_->begin());
+  }
+
+  ST zr{0.};
+  ST zi{0.};
+  ST wr{0.};
+  ST wi{0.};
+  ST newval_r{0.};
+  ST newval_i{0.};
+  for (auto i = 0; i < basis_set_size; i++)
+  {
+    for (auto j = 0; j < OrbitalSetSize; j++)
+    {
+      // cur_elem points to the real componend of the coefficient.
+      // Imag component is adjacent in memory.
+      const auto cur_elem = Nsplines * i + 2 * j;
+      zr                  = 0.;
+      zi                  = 0.;
+      wr                  = 0.;
+      wi                  = 0.;
+      newval_r            = 0.;
+      newval_i            = 0.;
+      for (auto k = 0; k < OrbitalSetSize; k++)
+      {
+        const auto index = Nsplines * i + 2 * k;
+        zr               = (*coef_copy_)[index];
+        zi               = (*coef_copy_)[index + 1];
+        wr               = rot_mat[k][j].real();
+        wi               = rot_mat[k][j].imag();
```

Comment:
zri, wri are both used local. Could you move the definition here.

Reply:
Done.